portfolio/src/main/webapp/script.js

```diff
@@ -45,8 +45,14 @@ function toggleVisibility (sectionToToggle) {
   }
 }
 
-async function fetchGreeting() {
+async function fetchComments() {
   const response = await fetch('/data');
-  const greeting = await response.text();
-  document.getElementById('fetched-greeting').innerHTML = greeting;
+  const comments = await response.json();
+  const commentsSection = document.getElementById('comments');
+  commentsSection.innerHTML = '';
+  for (var i = 0; i < comments.length; i++) {
```

Comment:
super nit: optional, but you can also iterate through the array by element (instead of index) which is slightly more readable. ex: for (const comment in comments) .... Otherwise it's slightly better to use a better index variable name like "commentsIndex" instead of "i".

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in